test/health_check_test_case.py

```diff
@@ -259,22 +259,26 @@ def _expected_elasticsearch(self, up: bool) -> JSON:
         }
 
     def _test(self, endpoint_states: Mapping[str, bool], lambdas_up: bool, path: str = '/health/fast'):
-        with ResponsesHelper() as helper:
+        with responses.RequestsMock() as helper:
             helper.add_passthru(self.base_url)
             self._mock_other_lambdas(helper, lambdas_up)
             with self._mock_service_endpoints(helper, endpoint_states):
                 return requests.get(self.base_url + path)
 
     @contextmanager
-    def _mock_service_endpoints(self, helper: ResponsesHelper, endpoint_states: Mapping[str, bool]) -> None:
+    def _mock_service_endpoints(self,
+                                helper: responses.RequestsMock,
+                                endpoint_states: Mapping[str, bool]) -> None:
+        helper.assert_all_requests_are_fired = False
         for endpoint, endpoint_up in endpoint_states.items():
             helper.add(responses.Response(method='HEAD',
                                           url=config.service_endpoint() + endpoint,
                                           status=200 if endpoint_up else 503,
                                           json={}))
         yield
 
-    def _mock_other_lambdas(self, helper: ResponsesHelper, up: bool):
+    def _mock_other_lambdas(self, helper: responses.RequestsMock, up: bool):
+        helper.assert_all_requests_are_fired = False
```

Comment:
A comment should explain the intent. I usually answer the assumed question "What would happen if this weren't there?"